ming_sim/rescript_draft.py

```
@@ -155,9 +158,23 @@ def normalize_rescript_layer_a_option(raw: object) -> Dict[str, object]:
     # amount 传 raw 原值；用返回值写 out["amount"]（honorific 无则不写）。
+    # grant_kind=army_pay → 内部 grant_action=协饷；矛盾 typed shape 响亮拒绝；
+    # 不按 target_kind/label 升格，不补 purpose/target（#1503 五字段仍须显式）。
     if action_type == "grant_allocation":
         from ming_sim.action_materialize import require_grant_allocation_shape
 
+        grant_kind = ""
+        if "grant_kind" in raw and raw["grant_kind"] is not None:
+            grant_kind = str(raw["grant_kind"]).strip()
+        if grant_kind:
+            if grant_kind != _GRANT_KIND_ARMY_PAY:
+                raise ValueError(f"grant 非法 grant_kind：{grant_kind!r}")
```

Ruff (0.16.2)

[warning] String contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF001)